sdk/synapse/synapse-spark/src/SparkClient.ts

```diff
@@ -0,0 +1,481 @@
+
+import { SynapseSpark } from "./generated";
+import { logger } from "./logger";
+import { SDK_VERSION, DEFAULT_SYNAPSE_SCOPE } from "./constants";
+import { createSpan, getCanonicalCode } from "./tracing";
+
+import {
+  SparkClientOptions,
+  GetSparkBatchJobOptions,
+  ListSparkBatchJobsOptions,
+  CreateSparkBatchJobOptions,
+  CancelSparkBatchJobOptions,
+  GetSparkSessionOptions,
+  ListSparkSessionsOptions,
+  CreateSparkSessionOptions,
+  CancelSparkSessionOptions,
+  ResetSparkSessionTimeoutOptions,
+  GetSparkStatementOptions,
+  ListSparkStatementsOptions,
+  CreateSparkStatementOptions,
+  CancelSparkStatementOptions,
+  SparkBatchJobOptions,
+  SparkSessionOptions,
+  SparkStatementOptions,
+  OperationResponse
+} from "./models";
+
+import {
+  GetSparkBatchJobResponse,
+  ListSparkBatchJobsResponse,
+  CreateSparkBatchJobResponse,
+  GetSparkSessionResponse,
+  ListSparkSessionsResponse,
+  CreateSparkSessionResponse,
+  GetSparkStatementResponse,
+  ListSparkStatementsResponse,
+  CreateSparkStatementResponse
+} from "./models";
+
+export { PipelineOptions, logger };
+
+export class SparkClient {
+  /**
+   * The base URL to the workspace
+   */
+  public readonly workspaceEndpoint: string;
+
+  /**
+   * @internal
+   * @ignore
+   * A reference to the auto-generated synapse spark HTTP client.
+   */
+  private readonly client: SynapseSpark;
+
+  constructor(
+    workspaceEndpoint: string,
+    sparkPoolName: string,
+    credential: TokenCredential,
+    pipelineOptions: SparkClientOptions = {}
+  ) {
+    this.workspaceEndpoint = workspaceEndpoint;
+
+    const libInfo = `azsdk-js-synapse-spark/${SDK_VERSION}`;
+
+    const userAgentOptions = pipelineOptions.userAgentOptions;
+
+    pipelineOptions.userAgentOptions = {
+      ...pipelineOptions.userAgentOptions,
+      userAgentPrefix:
+        userAgentOptions && userAgentOptions.userAgentPrefix
+          ? `${userAgentOptions.userAgentPrefix} ${libInfo}`
+          : libInfo
+    };
+
+    const authPolicy = bearerTokenAuthenticationPolicy(credential, DEFAULT_SYNAPSE_SCOPE);
+
+    const internalPipelineOptions = {
+      ...pipelineOptions,
+      ...{
+        loggingOptions: {
+          logger: logger.info
+        }
+      }
+    };
+
+    const pipeline = createPipelineFromOptions(internalPipelineOptions, authPolicy);
+
+    this.client = new SynapseSpark(credential, workspaceEndpoint, sparkPoolName, pipeline);
+  }
+
+  /**
+   * Gets a single spark batch job.
+   * @param batchId Identifier for the batch job.
+   * @param options The options parameters.
+   */
+
+  public async getSparkBatchJob(
+    batchId: number,
+    options: GetSparkBatchJobOptions = {}
+  ): Promise<GetSparkBatchJobResponse> {
+    const { span, updatedOptions } = createSpan("Synapse-GetSparkBatchJob", options);
+
+    try {
+      const response = await this.client.sparkBatch.getSparkBatchJob(
+        batchId,
+        operationOptionsToRequestOptionsBase(updatedOptions)
```

Comment:
you don't need to use `operationOptionsToRequestOptionsBase` because your generated layer is already accepting `OperationOptions`. Remove from all client operation methods.

Reply:
Thanks for comments Will remove that.

---
In reply to: [521614448](https://github.com/Azure/azure-sdk-for-js/pull/12487#discussion_r521614448) [](ancestors = 521614448)